### DISCUSSION TITLE
Generate mutiple prisma client for multiple schemas of the same postgres database

### DISCUSSION THREAD

**Discussion by khalilTN** [prisma/prisma#6793] [Q&A] [answered]
Hello,

Before all thanks for your awsome work.
Actually, we are trying to connect to existing postgres database containing 3 different schemas.
We generated one client for each schema (db pull).
Accessing the first schema is performed without troubles, but the for the sceond one, it seems like its client is always trying to access the first schema, even if the table name is correct.
It seems like the first made connection is always preserved.

Can you clarify !
Thanks

**Comment by janpio** [comment]
How are you generating the schemas? 
How are you creating your `PrismaClient` instances in code?

**Reply by khalilTN** [reply]
Im génération schémas via `prisma db pull` and `prisma generate`. I instanciate the client that i need in code.

**Reply by janpio** [reply]
That makes sense.

How are you going about the _multiple_ Prisma Clients? 
Where do you store the schema files in your project? What command do you use to generate the _3_ Prisma Clients? How do you use all _3_ in code?

**Reply by khalilTN** [reply]
Actually each one is in a distinct folder and has its own schema file (it's own connection string). I execute db pull/generate for each one of the schemas.
In my business layer, i instanciate the client needed.

**Reply by janpio** [reply]
Do you define the connection string in code, or hard code it in your schema? 
If not, it is probably reading the same environment variable for all `PrismaClient`s you are instantiating.

**Reply by khalilTN** [reply]
In each folder i have a .env file containing the connection string of the schema.
Db pull et generate are working perfectly.
The error that i got is `trying to access table schema1.table1 that does not exist in schema`. 
Table1 exists in schema 2 and not schema1.

**Reply by janpio** [reply]
That could be it: the `.env` file is taken from your application (which you only have 1) and not from where you load the Prisma Client from. That could explain them all using the same schema.

You need to supply the connection string when you instantiate the `PrismaClient` in your application: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference/#examples

**Comment by khalilTN** [comment, marked as answer]
@janpio Thanks, that was the issue, i think .env is read only one time, that's the cause.
I'm passing connection string in constructor, everything is fine now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
